components/match2/commons/opponent.lua

```diff
@@ -289,19 +289,30 @@ function Opponent.readOpponentArgs(args)
 		}
 
 	elseif partySize == 1 then
+		local playerTeam = args.team or args.p1team
+		if playerTeam then
+			playerTeam = playerTeam:lower():gsub('_', ' ')
+			playerTeam = TeamTemplate.resolve(playerTeam, date)
```

Comment:
`TeamTemplate.resolve()` is pretty expensive and shouldn't be performed while reading input. maybe move to `Opponent.resolve()` with the rest of the resolves?